third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -1152,6 +1152,48 @@ impl<'env> Generator<'env> {
         }
     }
 
+    /// Generate borrow_field when unpacking a reference to a struct
+    // e.g. `let s = &S; let (a, b, c) = &s`, a, b, and c are references
+    fn gen_borrow_field_for_unpack_ref(
+        &mut self,
+        id: &NodeId,
+        str: &QualifiedInstId<StructId>,
+        arg: TempIndex,
+        temps: Vec<TempIndex>,
+        ref_kind: ReferenceKind,
+    ) {
+        let struct_env = self.env().get_struct(str.to_qualified_id());
+        let mut temp_filed_offsets = BTreeMap::new();
+        for (field, input_temp) in struct_env.get_fields().zip(temps.clone()) {
```

Comment:
You don't actually need `temp_filed_offsets` (or `temp_to_field_offsets`), as field names should already be unique, and you only use it once:
```
for (field_offset, temp) in struct_env.get_fields().zip(temps.clone()) {
    self.with-reference_mode(|s, entering| {
        ...
```

Reply:
@brmataptos, I have tried using one loop to do it but there was a borrow issue.